src/Wrapper/GenericAKVWrapper.php

```diff
@@ -188,12 +226,39 @@ public function decrypt(string $encryptedData): string
         } catch (Throwable $e) {
             throw new ApplicationException('Deciphering failed.', $e->getCode(), $e);
         }
-        $this->verifyMetadata($decryptedContext[self::METADATA_INDEX], $this->metadata);
+        $this->verifyMetadata($decryptedContext[self::METADATA_INDEX], $metadata);
         try {
             $key = Key::loadFromAsciiSafeString($decryptedContext[self::KEY_INDEX]);
             return Crypto::decrypt($encrypted[self::PAYLOAD_INDEX], $key, true);
         } catch (Throwable $e) {
+            $doBackfill = false;
             throw new ApplicationException('Deciphering failed.', $e->getCode(), $e);
+        } finally {
+            if ($doBackfill) {
```

Comment:
jako tohle s tím finally mi dalo trošku zabrat - nejde to napsat nějak takhle?

   try {
            $key = Key::loadFromAsciiSafeString($decryptedContext[self::KEY_INDEX]);
            $decrypted = Crypto::decrypt($encrypted[self::PAYLOAD_INDEX], $key, true);
            if ($dobackfill) {
                $this->backfill(...) 
           } 
            return $decrypted;
        } catch (Throwable $e) {

Reply:
Jo, šlo by, ale přes `finally` mi přišlo elegantní, že ten backfill tolik nezasahuje do původního code flow. 🙂

Reply:
asi jo no, snad to tam nebude dlouho